projects/ngx-formentry/src/form-entry/value-adapters/obs-adapter-helper.ts

```diff
@@ -489,20 +489,39 @@ export class ObsAdapterHelper {
   getGroupPayload(node: NodeBase) {
     const nodeAsGroup: GroupNode = node as GroupNode;
 
-    let childrenPayload = [];
+    // Get existing obs
+    let childrenPayload =
+      nodeAsGroup?.initialValue?.groupMembers?.map((node) =>
+        this.getOldObsPayload(node)
+      ) || [];
+
+    let isGroupChanged: boolean = false;
     _.each(nodeAsGroup.children, (child) => {
       const payload = this.getObsNodePayload(child);
       if (payload.length > 0) {
-        childrenPayload = childrenPayload.concat(payload);
+        isGroupChanged = true;
+        if (payload[0].uuid) {
```

Comment:
This assumes that if `payload` has a `uuid` that `childrenPayload` has a obs with that uuid, otherwise the obs is dropped. Also, given that the `payload` might itself be a grouped obs, I think we need to iterate across all elements in payload rather than just assuming `payload[0]` is the only thing that matters.